CI fails frequently after introducing `pytest-xdist`

### Expected behavior

I will update the issue description whenever I find another failure.

In principle, the CI sometimes fails due to the parallel run by the following PR: 
- https://github.com/optuna/optuna/pull/6170

### Environment

CI environment

### Error messages, stack traces, or logs

```shell
- https://github.com/optuna/optuna/actions/runs/16332163798/job/46137014991#step:10:426


_________________ test_log_completed_trial_skip_storage_access _________________
[gw4] linux -- Python 3.9.23 /opt/hostedtoolcache/Python/3.9.23/x64/bin/python

    def test_log_completed_trial_skip_storage_access() -> None:
        study = create_study()
    
        # Create a trial to retrieve it as the `study.best_trial`.
        study.optimize(lambda _: 0.0, n_trials=1)
        frozen_trial = study.best_trial
    
        storage = study._storage
    
        with patch.object(storage, "get_best_trial", wraps=storage.get_best_trial) as mock_object:
            study._log_completed_trial(frozen_trial.values, frozen_trial.number, frozen_trial.params)
>           assert mock_object.call_count == 1
E           AssertionError: assert 0 == 1
E            +  where 0 = <MagicMock name='get_best_trial' id='140017043141632'>.call_count

tests/study_tests/test_study.py:1154: AssertionError


- https://github.com/optuna/optuna/actions/runs/16327000266/job/46119643469#step:12:301


____ test_get_timeline_plot_with_killed_running_trials[0.0-plot_timeline0] _____
[gw7] linux -- Python 3.8.18 /opt/hostedtoolcache/Python/3.8.18/x64/bin/python

plot_timeline = <function plot_timeline at 0x7ff526abbb80>, waiting_time = 0.0

    @parametrize_plot_timeline
    @pytest.mark.parametrize("waiting_time", [0.0, 1.5])
    def test_get_timeline_plot_with_killed_running_trials(
        plot_timeline: Callable[..., Any], waiting_time: float
    ) -> None:
        def _objective_with_sleep(trial: optuna.Trial) -> float:
            sleep_start_datetime = datetime.datetime.now()
            # Spin waiting is used here because high accuracy is necessary even in weak VM.
            # Please check the motivation of the bugfix in https://github.com/optuna/optuna/pull/5549/
            while datetime.datetime.now() - sleep_start_datetime < datetime.timedelta(seconds=0.1):
                # `sleep(0.1)` is only guaranteed to rest for more than 0.1 second; the actual time
                # depends on the OS. spin waiting is used here to rest for 0.1 second as precisely as
                # possible without voluntarily releasing the context.
                pass
            assert datetime.datetime.now() - sleep_start_datetime < datetime.timedelta(seconds=0.19)
            trial.suggest_float("x", -1.0, 1.0)
            return 1.0
    
        study = optuna.create_study()
        trial = optuna.trial.create_trial(
            params={"x": 0.0},
            distributions={"x": optuna.distributions.FloatDistribution(-1.0, 1.0)},
            value=None,
            state=TrialState.RUNNING,
        )
        study.add_trial(trial)
>       study.optimize(_objective_with_sleep, n_trials=2)

tests/visualization_tests/test_timeline.py:221: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/optuna/study/study.py:490: in optimize
    _optimize(
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/optuna/study/_optimize.py:63: in _optimize
    _optimize_sequential(
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/optuna/study/_optimize.py:160: in _optimize_sequential
    frozen_trial_id = _run_trial(study, func, catch)
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/optuna/study/_optimize.py:258: in _run_trial
    raise func_err
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/optuna/study/_optimize.py:201: in _run_trial
    value_or_values = func(trial)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

trial = <optuna.trial._trial.Trial object at 0x7ff523f32d00>

    def _objective_with_sleep(trial: optuna.Trial) -> float:
        sleep_start_datetime = datetime.datetime.now()
        # Spin waiting is used here because high accuracy is necessary even in weak VM.
        # Please check the motivation of the bugfix in https://github.com/optuna/optuna/pull/5549/
        while datetime.datetime.now() - sleep_start_datetime < datetime.timedelta(seconds=0.1):
            # `sleep(0.1)` is only guaranteed to rest for more than 0.1 second; the actual time
            # depends on the OS. spin waiting is used here to rest for 0.1 second as precisely as
            # possible without voluntarily releasing the context.
            pass
>       assert datetime.datetime.now() - sleep_start_datetime < datetime.timedelta(seconds=0.19)
E       AssertionError: assert (datetime.datetime(2025, 7, 16, 18, 15, 55, 98863) - datetime.datetime(2025, 7, 16, 18, 15, 54, 907173)) < datetime.timedelta(microseconds=190000)
E        +  where datetime.datetime(2025, 7, 16, 18, 15, 55, 98863) = <built-in method now of type object at 0x7ff52ea36d40>()
E        +    where <built-in method now of type object at 0x7ff52ea36d40> = <class 'datetime.datetime'>.now
E        +      where <class 'datetime.datetime'> = datetime.datetime
E        +  and   datetime.timedelta(microseconds=190000) = <class 'datetime.timedelta'>(seconds=0.19)
E        +    where <class 'datetime.timedelta'> = datetime.timedelta

tests/visualization_tests/test_timeline.py:209: AssertionError
```

### Steps to reproduce

Run the CI many times.
Some tests fail stochastically.

### Additional context (optional)

_No response_

--- a/tests/samplers_tests/test_brute_force.py
+++ b/tests/samplers_tests/test_brute_force.py
@@ -326,7 +326,8 @@ def test_not_avoid_premature_stop() -> None:
         #   trials[1]: x = 1         # Running
         #   trials[2]: x = 0, y = 0  # Running
         #
-        # Assume that all possible combinations of x and y have been exhausted.
+        # Since the sampler assumes that running trials already suggest all parameters, the sampler
+        # considers all possible combinations of x and y have been exhausted.
         study.tell(trials[1], 0.0)
         study.tell(trials[2], 0.0)
         assert mock_stop.call_count == 2
@@ -356,8 +357,9 @@ def test_avoid_premature_stop() -> None:
         #   trials[1]: x = 1         # Running
         #   trials[2]: x = 0, y = 0  # Running
         #
-        # If `avoid_premature_stop` is set to `True`, the sampler should not stop,
-        # because BruteForceSampler assumes that `y` for trials[1] has not been suggested yet
+        # If `avoid_premature_stop` is `True`, the sampler assumes that running trials may still
+        # suggest new parameters, considering the possibility for trials[1] to suggest `y`.
+        # So the sampler should not stop.
         study.tell(trials[2], 0.0)
         mock_stop.assert_not_called()
 
@@ -387,6 +389,6 @@ def test_avoid_premature_stop() -> None:
         #   trials[2]: x = 0, y = 0  # Completed
         #   trials[3]: x = 1, y = 1  # Running
         #
-        # Assume that all possible combinations of x and y have been exhausted.
+        # All possible combinations of x and y have been exhausted.
         study.tell(trials[3], 0.0)
         mock_stop.assert_called_once()

